caddytls/config.go

```diff
@@ -394,16 +394,9 @@ func assertConfigsCompatible(cfg1, cfg2 *Config) error {
 	if c1.MaxVersion != c2.MaxVersion {
 		return fmt.Errorf("maximum TLS version mismatch")
 	}
-	if c1.ClientAuth != c2.ClientAuth {
-		return fmt.Errorf("client authentication policy mismatch")
-	}
-	if c1.ClientAuth != tls.NoClientCert && c2.ClientAuth != tls.NoClientCert && c1.ClientCAs != c2.ClientCAs {
-		// Two hosts defined on the same listener are not compatible if they
-		// have ClientAuth enabled, because there's no guarantee beyond the
-		// hostname which config will be used (because SNI only has server name).
-		// To prevent clients from bypassing authentication, require that
-		// ClientAuth be configured in an unambiguous manner.
```

Comment:
Can you preserve the comment (and update it and move it as needed)? It's a little confusing what is going on in the changes otherwise.

Reply:
Sure, I'll put it back. To be honest I don't quite get what this comment means here.

Reply:
Added it back.

Reply:
I guess it does a bad job of explaining, but basically: if you have two sites defined in the Caddyfile: `example.com/foo` and `example.com/bar`, and one has ClientAuth but the other does not, the expected behavior is ambiguous since we do not know the path at TLS-handshake-time, thus it is possible that a config not requiring client auth would be chosen even if it is needed, which is bad.

Reply:
@mholt: Thanks for the explanation. It's got more clear with your example!